cpp/include/cuvs/neighbors/cagra.h

```diff
@@ -334,6 +334,38 @@ cuvsError_t cuvsCagraIndexDestroy(cuvsCagraIndex_t index);
  */
 cuvsError_t cuvsCagraIndexGetDims(cuvsCagraIndex_t index, int* dim);
 
+/**
+ * @}
+ */
+
+/**
+ * @defgroup cagra_c_merge_params C API for CUDA ANN Graph-based nearest neighbor search
+ * @{
+ */
+
+/**
+ * @brief Supplemental parameters to merge CAGRA index
+ *
+ */
+/** Strategy for merging CAGRA indices. */
+typedef enum {
```

Comment:
Can we reuse this in the C++ version so we don't have to duplicate?

Reply:
No, I'm afraid we can't because C doesn't support namespaces, `::` , and `enum class`. It's not first we redefine in C which should work well, like, [`hash_mode`](https://github.com/rapidsai/cuvs/blob/branch-25.06/cpp/include/cuvs/neighbors/cagra.hpp#L173) vs [`cuvsCagraHashMode`](https://github.com/rapidsai/cuvs/blob/branch-25.06/cpp/include/cuvs/neighbors/cagra.h#L224)

Reply:
> We do this with the the "DistanceType` in pairwise distances. Maybe we can do the same thing here?

Sorry, I found we defined the separate [DistanceType](https://github.com/rapidsai/cuvs/blob/branch-25.06/cpp/include/cuvs/distance/distance.h#L69C3-L69C19) and converted it to cpp type via [`static_cast`](https://github.com/rapidsai/cuvs/blob/branch-25.06/cpp/src/neighbors/cagra_c.cpp#L45). Maybe I found a wrong position. Could you hint me at the code position by any chance, many thanks?

Reply:
@rhdong, [this](https://github.com/rapidsai/cuvs/blob/7affbc034e36cb66288391302ba6f910b54cd517/cpp/include/cuvs/distance/distance.hpp#L29) is where we map the C++ type to the C type. That static cast shouldn't be needed and I suspect it's an artifact from before we have formally combined them. You are free to fix it if you have time.

Reply:
Oh, I understood! Let C++ be compatible with C :smiley: